### DISCUSSION TITLE
Question regarding inMaxBodies when initializing the PhysicsSystem

### DISCUSSION THREAD

**Discussion by Pindrought** [jrouwe/JoltPhysics#917] [Q&A] [answered]
I noticed some weird behavior, and realized it was occurring because my max bodies was only set to 1,024. I have since increased it to 65535 as suggested in the comments.

I am wondering what are the implications of increasing the number here?

I see that there is a vector that gets reserved for the bodies in the body manager as well as two arrays get allocated for rigid and soft bodies in the body manager.
I see that there is a vector generated for the sequence numbers for bodies in the body manager.
Lastly, I see that in the island builder there is an array allocated for the BodyLinks.

**If I am to increase the number of max bodies, is there any performance impact or is it only just resulting in more memory being allocated?** 

Also, is there any way to adjust the parameters after the physics system has been initialized or does it need to be completely shut down and reinitialized to do so? 
Can the physics system even be shut down? I saw that the PhysicsSystem has a destructor in place, but the IslandBuilder's destructor didn't look like it was implemented.

As always thank you very much! I'm loving Jolt so far.

**Comment by jrouwe** [comment, marked as answer]
> I am wondering what are the implications of increasing the number here?

Costs more memory (but not a lot).

> Also, is there any way to adjust the parameters after the physics system has been initialized or does it need to be completely shut down and reinitialized to do so?

It needs to be shut down completely. The reason for preallocating all of this is that Jolt allows multi-threaded access to many of its structures, reallocating would break this.

> Can the physics system even be shut down? I saw that the PhysicsSystem has a destructor in place, but the IslandBuilder's destructor didn't look like it was implemented.

Yes it can, the IslandBuilder only uses temporary allocations that are freed every frame. It doesn't need a destructor.

**Reply by Pindrought** [reply]
Awesome, thanks! Also, it looks like I was wrong about the IslandBuilder destructor. My intellisense can't find the destructor for some reason, but it's there.

**Reply by jrouwe** [reply]
Ah, I misremembered, there is a bit of memory it keeps from frame to frame.
